plugins-base/XSFeatureSRInfo.py

```diff
@@ -112,7 +112,7 @@ def MenuRegenerator(cls, inList, inMenu):
         srList = [ sr for sr in HotAccessor().visible_sr if sr.other_config({}).get('xensource_internal', '') != 'true' ]
 
         # Sort list by SR shared flag then name
-        srList.sort(lambda x, y: cmp(y.shared(False), x.shared(False)) or cmp (x.name_label(''), y.name_label()))
+        srList.sort(key=lambda sr: (-sr.shared(False), sr.name_label('')))
```

Comment:
is `-sr` right here?

Personally, I think the logic would be easier to follow as

```python
# Sort list by SR shared flag then name
srList.sort(key=lambda sr: sr.name_label(''))
srList.sort(key=lambda sr: sr.shared(False))
```
maybe with a `reverse=True` in the second line if I've interpreted the intent of the negation properly ?

Reply:
Alternatively, `srList.sort(key=lambda sr: (not sr.shared(False), sr.name_label('')))` might be better ?  At least it makes clear that the first part of the tuple is supposed to be a boolean.

Tangent, passing the default-if-missing value in makes this code horrible to read and follow.  Do you happen to know if this is a local artifact, or a property of the API?

Reply:
> is -sr right here?
Yes, it is correct. The 1st half of the original `cmp(y.shared(False), x.shared(False))` had y and x reversed, so the primary sort order is reversed.

Splitting the nested sort into two consecutive sorts would make the result reliant of the sort algorithm chosen by the Python interpreter for the given data and I'd like the result to be kept clearly defined as a nested sort with two sort keys. Therefore, I see the change is correct!

Because the sort is already commented using a meaningful comment, the sort does not need further explanation:
```py
# Sort list by SR shared flag then name
srList.sort(key=lambda sr: (-sr.shared(False), sr.name_label('')))
```
Even though the comment says that `sr.shared(False)` returns a flag, nothing here ensures that it will always return a boolean. In principle, it could return anything (numbers, floats or a mix of all that). Therefore I think keeping `-sr.shared(False)` is the safe thing to do (does not change the behavior at all) and it's clear that `-` means negation.